app/controllers/user_notifications_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UserNotificationsController < Api::V2::BaseController
+  before_action :require_login
+  before_action :find_resource, :only => [:update, :destroy]
+
+  def index
+    @notifications = NotificationRecipient.
+    where(:user_id => User.current.id).
+    order(:created_at).
+    eager_load(:notification, :notification_type)
+
+    render :json => {
+      :notifications => @notifications.paginate(paginate_options).map(&:payload),
+      :total => @notifications.count
+    }
+  end
+
+  def update
```

Comment:
Let's keep these for next PRs, I'd really prefer for this to not go over the current LOC - you can always submit a PR based on this branch as https://github.com/theforeman/foreman/pull/4170 did